sup3r/bias/bias_transforms.py

```diff
@@ -488,35 +492,62 @@ def local_qdm_bc(data: np.array,
     >>> unbiased = local_qdm_bc(biased_array, lat_lon_array, "ghi", "rsds",
     ...                         "./dist_params.hdf")
     """
+    # Confirm that the given time matches the expected data size
+    assert (
+        data.shape[2] == time.size
+    ), 'Time should align with data 3rd dimension'
+
     base, bias, bias_fut, cfg = get_spatial_bc_quantiles(lat_lon,
                                                          base_dset,
                                                          feature_name,
                                                          bias_fp,
                                                          threshold)
+
     if lr_padded_slice is not None:
         spatial_slice = (lr_padded_slice[0], lr_padded_slice[1])
         base = base[spatial_slice]
         bias = bias[spatial_slice]
         bias_fut = bias_fut[spatial_slice]
 
-    if no_trend:
-        mf = None
-    else:
-        mf = bias_fut.reshape(-1, bias_fut.shape[-1])
-    # The distributions are 3D (space, space, N-params)
-    # Collapse 3D (space, space, N) into 2D (space**2, N)
-    QDM = QuantileDeltaMapping(base.reshape(-1, base.shape[-1]),
-                               bias.reshape(-1, bias.shape[-1]),
-                               mf,
-                               dist=cfg['dist'],
-                               relative=relative,
-                               sampling=cfg["sampling"],
-                               log_base=cfg["log_base"])
-
-    # input 3D shape (spatial, spatial, temporal)
-    # QDM expects input arr with shape (time, space)
-    tmp = data.reshape(-1, data.shape[-1]).T
-    # Apply QDM correction
-    tmp = QDM(tmp)
-    # Reorgnize array back from  (time, space) to (spatial, spatial, temporal)
-    return tmp.T.reshape(data.shape)
+    output = np.full_like(data, np.nan)
+    idx = [
+        np.argmin(abs(d - cfg['time_window_center'])) for d in time.day_of_year
+    ]
+    for i in set(idx):
+        # Naming following the paper: observed historical
+        oh = base[:, :, i]
+        # Modeled historical
+        mh = bias[:, :, i]
+        # Modeled future
+        mf = bias_fut[:, :, i]
+
+        # This satisfies the rex's QDM design
+        if no_trend:
+            mf = None
+        else:
+            mf = mf.reshape(-1, mf.shape[-1])
+        # The distributions are 3D (space, space, N-params)
+        # Collapse 3D (space, space, N) into 2D (space**2, N)
```

Comment:
Can you update these comments for the new output shape?

Reply:
At this point, after applied the `window_idx` in the inputs, these arrays have the same shape and dimensions, so nothing changed here. I'm adding a few more comments anyways.

Reply:
I see, not here, but in the documentation. It was updated.